arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -149,18 +141,11 @@ std::shared_ptr<LogicalCollection> CollectionNameResolver::getCollectionStructCl
     return getCollectionStruct(name);
   }
 
-  try {
-    // We have to look up the collection info:
-    ClusterInfo* ci = ClusterInfo::instance();
-    auto cinfo = ci->getCollection(_vocbase.name(), name);
-
-    TRI_ASSERT(cinfo != nullptr);
-
-    return cinfo;
-  } catch (...) {
-  }
+  // We have to look up the collection info:
+  ClusterInfo* ci = ClusterInfo::instance();
+  auto cinfo = ci->getCollectionNT(_vocbase.name(), name);
```

Comment:
can directly return here instead of assigning the collection to a local variable.